Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/TestFolderIntegrationTests.cs

```diff
@@ -33,24 +33,28 @@ public void TearDown()
 		[Test]
 		public void GetFolderName()
 		{
-			// Arrange
 			string _workspaceName = $"IntTest_{Guid.NewGuid()}";
 			IServicesMgr servicesManager = testHelper.GetServicesManager();
 			int _workspaceId = WorkspaceHelpers.CreateWorkspace.CreateWorkspaceAsync(_workspaceName,
 				SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME, servicesManager,
 				SharedTestHelpers.ConfigurationHelper.ADMIN_USERNAME, SharedTestHelpers.ConfigurationHelper.DEFAULT_PASSWORD).Result;
-			int rootFolderArtifactId = FoldersHelper.GetRootFolderArtifactID(_workspaceId, servicesManager,
-				ConfigurationHelper.ADMIN_USERNAME, ConfigurationHelper.DEFAULT_PASSWORD);
-			
-			// Act
-			// Get Folder Name
-			string folderName = SuT.GetFolderName(rootFolderArtifactId, _workspaceId);
+			try
+			{
+				// Arrange
+				int rootFolderArtifactId = FoldersHelper.GetRootFolderArtifactID(_workspaceId, servicesManager,
+					ConfigurationHelper.ADMIN_USERNAME, ConfigurationHelper.DEFAULT_PASSWORD);
 
-			// Assert
-			Assert.AreEqual(_workspaceName, folderName);
+				// Act
+				string folderName = SuT.GetFolderName(rootFolderArtifactId, _workspaceId);
 
-			//Delete Workspace
-			WorkspaceHelpers.DeleteWorkspace.DeleteTestWorkspace(_workspaceId, servicesManager, ConfigurationHelper.ADMIN_USERNAME, ConfigurationHelper.DEFAULT_PASSWORD);
+				// Assert
+				Assert.AreEqual(_workspaceName, folderName);
+			}
+			finally
+			{
+				//Delete Workspace
+				WorkspaceHelpers.DeleteWorkspace.DeleteTestWorkspace(_workspaceId, servicesManager, ConfigurationHelper.ADMIN_USERNAME, ConfigurationHelper.DEFAULT_PASSWORD);
```

Comment:
also before deleting the workspace, we should query if it exists then proceed with the delete.

Reply:
Moved Create Workspace to Setup and Delete Workspace to Teardown so that Delete Workspace will only run if the Setup passes